### PR TITLE
escape $$ in asciidoc when generating api-reference pages

### LINKED ISSUE #17155: API Reference: shows "(VAR_NAME)" instead of "$$(VAR_NAME)"
In <http://kubernetes.io/v1.1/docs/api-reference/v1/definitions.html#_v1_container> it reads `The $(VAR_NAME) syntax can be escaped with a double , ie: (VAR_NAME).` but it should read `The $(VAR_NAME) syntax can be escaped with a double $$, i.e. $$(VAR_NAME).`

### PR DESCRIPTION
**Pull request by caesarxuchao**
Fix #17155 on head.

cc @orfjackal

### CODE REVIEW

File: hack/update-api-reference-docs.sh [hunk 1]
```diff
@@ -50,11 +50,11 @@ SWAGGER_PATH="${REPO_DIR}/api/swagger-spec/"
 
 echo "Reading swagger spec from: ${SWAGGER_PATH}"
 
-docker run -u $(id -u) --rm -v $V1_TMP_IN_HOST:/output:z -v ${SWAGGER_PATH}:/swagger-source:z gcr.io/google_containers/gen-swagger-docs:v4 \
+docker run -u $(id -u) --rm -v $V1_TMP_IN_HOST:/output:z -v ${SWAGGER_PATH}:/swagger-source:z gcr.io/google_containers/gen-swagger-docs:v4.1 \
```
Comment (caesarxuchao) [inline hack/update-api-reference-docs.sh]: Just pushed the v4.1 image to gcr.io.

File: hack/gen-swagger-doc/gen-swagger-docs.sh [hunk 2]
```diff
@@ -53,6 +53,9 @@ sed -i -e 's|<<any>>|link:definitions.html#_any[any]|g' ./paths.adoc
 # change the title of paths.adoc from "paths" to "operations"
 sed -i 's|== Paths|== Operations|g' ./paths.adoc
 
+# $$ has special meaning in asciidoc, we need to escape it
+sed -i 's|\$\$|+++$$+++|g' ./definitions.adoc
```
Comment (nikhiljindal) [inline hack/gen-swagger-doc/gen-swagger-docs.sh]: Who does the unescaping? (Removing +++)

**Comment by nikhiljindal** [comment]
LGTM, modulo a clarifying question

File: hack/gen-swagger-doc/gen-swagger-docs.sh (hunk 2 above)
Comment (caesarxuchao) [inline hack/gen-swagger-doc/gen-swagger-docs.sh]: asciidoctor does that 2 lines later.

**Comment by caesarxuchao** [comment]
@k8s-bot test this please

**Comment by caesarxuchao** [comment]
@k8s-bot test this please

**Comment by caesarxuchao** [comment]
@k8s-bot test this please

**Comment by wojtek-t** [comment]
@caesarxuchao - please rebase

**Comment by caesarxuchao** [comment]
Adding back the LGTM after rebase. Thank you @wojtek-t for the reminder.

**Comment by nikhiljindal** [comment]
@caesarxuchao Jenkins is failing with `10:01:43 /workspace/kubernetes/docs/api-reference is out of date. Please run hack/update-api-reference-docs.sh`, can you please take a look

**Comment by caesarxuchao** [comment]
Thanks. It should be fixed now.

**Pull request merged**